driver/ppm_fillers.c

```diff
@@ -164,7 +154,7 @@ int f_sys_single(struct event_filler_arguments *args)
 	int res;
 	syscall_arg_t val;
 
-	syscall_get_arguments_deprecated(current, args->regs, 0, 1, &val);
+	syscall_get_arguments_deprecated(current, args, 0, 1, &val);
```

Comment:
Possibly stupid question: do we still need these if we have the args already in `args`? Or is it just a fancy way of saying `val = args->regs[0]`?

Reply:
Exactly that; moreover, `syscall_get_arguments_deprecated` also manage the copy of N params altogether.
Plus, i didn't want to push a much bigger change.
I think the new macro impl could become just a small loop, like:
```
for (i = start; i < nparams; i++) { out[i - start] = args[i]; }
```
I don't know if perf would be much better than using a small memcpy in the end though; i think it is good to go right now.

Reply:
I wouldn't be surprised if the generated code would be exactly identical. But also, if we now always have the syscall args handy, I'd rather directly refer to the copy already have rather than cleaning up the copy that's unnecessary in the first place.